src/Microsoft.AspNetCore.Mvc.Core/Builder/MvcEndpointInfo.cs

```diff
@@ -75,36 +75,36 @@ internal class MvcEndpointInfo
                     constraintBuilder.SetOptional(parameter.Name);
                 }
 
-                foreach (var inlineConstraint in parameter.InlineConstraints)
+                foreach (var inlineConstraint in parameter.Constraints)
                 {
-                    constraintBuilder.AddResolvedConstraint(parameter.Name, inlineConstraint.Constraint);
+                    constraintBuilder.AddResolvedConstraint(parameter.Name, inlineConstraint.Content);
                 }
             }
 
             return constraintBuilder.Build();
         }
 
         private static RouteValueDictionary GetDefaults(
-            RouteTemplate parsedTemplate,
+            RoutePattern parsedTemplate,
             RouteValueDictionary defaults)
         {
             var result = defaults == null ? new RouteValueDictionary() : new RouteValueDictionary(defaults);
```

Comment:
As defaults is already RVD, we can avoid wrapping it again. Also we can return early if defaults is null.